### PR TITLE
Update github formatting output to utilize groups

### LINKED ISSUE #421: Github format does not display file/line with errors
When utilizing the github format inside of a GitHub Action, the filename, and line number is not displayed in the GitHub Action UI.  This makes it challenging to resolve issues before you have opened an PR to view the annotations on the PR.  It would be great to add this information the the portion that is output in the action logs.

Currently the github format looks like this in raw text:

```
::error file=./myfile.yaml,line=14,col=22::[trailing-spaces] trailing spaces
```

The UI outputs:

```
Error: [trailing-spaces] trailing spaces
```

It would be great to update the output format to something like:

```
::error file=./myfile.yaml,line=14,col=22::./myfile.yaml:14:22: [trailing-spaces] trailing spaces
```

Or to utilize something like [log grouping](https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#grouping-log-lines) to provide a better structure similar to the default output format.

### PR DESCRIPTION
**Pull request by strangiato**
Resolves #421

Update the github formatting to utilize groups in the output and provide the line/column number for the error in the output log.

### CODE REVIEW

**Comment by adrienverge** [comment]
Thanks for your contribution @strangiato!

Could you provide screenshots of how it looked before, how it would look after?

@andrewimeson you contributed to the `github` format, do you approve this change?

**Comment by strangiato** [comment]
Let me mock up a repo to grab some screenshots.  Looks like some tests will need to be updated as well.

**Comment by strangiato** [comment]
Example run with Github actions:

https://github.com/strangiato/yamllint-demo/runs/4362632239?check_suite_focus=true

Screen Shot 2021-11-29 at 9 59 26 PM

**Comment by strangiato** [comment]
Tests have been updated and looks like everything is passing.

Let me know if you have any additional requests or feedback!

File: yamllint/cli.py [hunk 1]
```diff
@@ -129,6 +138,9 @@ def show_problems(problems, file, args_format, no_warn):
                 first = False
             print(Format.standard(problem, file))
 
+    if github_group:
```
Comment (adrienverge) [inline yamllint/cli.py:141]: Can we get rid of this variable `github_group` and simply use:
```suggestion
    if not first and args_format == 'github':
```
?

**Review by adrienverge — COMMENTED** [review]
Thanks @strangiato, the demo repo is helpful and the code looks good! (see my comment)

In [the original PR](https://github.com/adrienverge/yamllint/pull/322) that added support for `--format github`, annotations were added like this. How does your change integrates with this?
image

@andrewimeson last chance to say something? The implementation you wrote is about to change.

**Comment by andrewimeson** [comment]
I haven't tested it, but based on the documentation for the groups feature, I think this is a good enhancement and won't be a disruptive change. This will hopefully mitigate the UX issue that has caused multiple issue reports (#377, #364, and #357).

**Comment by andrewimeson** [comment]
As a nitpick, I would probably squash the three commits into one

**Review by adrienverge — APPROVED** [review]
> I haven't tested it, but based on the documentation for the groups feature, I think this is a good enhancement and won't be a disruptive change. This will hopefully mitigate the UX issue that has caused multiple issue reports (#377, #364, and #357).

@andrewimeson thanks for your constructive feedback!

> I updated the demo repo and setup a PR to show how it looks here:

It looks perfect.
For future reference, here is a screenshot of how it looks:
Capture d’écran de 2021-12-01 08-49-48

Thanks for your contribution!

**Pull request merged**